src/main/java/net/glowstone/entity/GlowLivingEntity.java

```diff
@@ -730,7 +739,8 @@ public void setHealth(double health) {
                 if (world.getGameRuleMap().getBoolean("doMobLoot")) {
                     LootData data = LootingManager.generate(this);
                     deathEvent.getDrops().addAll(data.getItems());
-                    if (data.getExperience() > 0) {
+                    // Only drop experience when hit by a player within 5 seconds (100 game ticks)
+                    if (ticksLived - playerDamageTick <= 100 && data.getExperience() > 0) {
```

Comment:
You'll probably want to have `playerDamageTick` default to -1 (in order to know if the entity wasn't hit yet), and add a check here for it being positive.

```java
if (playerDamageTick >= 0 && ticksLived - playerDamageTick <= 100 && data.getExperience() > 0)
```

Reply:
I made it ```private int playerDamageTick = -101;```, it solves the problem without extra code.